nixos/modules/services/network-filesystems/samba.nix

```
@@ -201,14 +201,14 @@ in
               message   = "If samba.nsswins is enabled, then samba.enableWinbindd must also be enabled";
             }
           ];
+      }
+
+      (mkIf cfg.enable {
         # Always provide a smb.conf to shut up programs like smbclient and smbspool.
         environment.etc."samba/smb.conf".source = mkOptionDefault (
           if cfg.enable then configFile
           else pkgs.writeText "smb-dummy.conf" "# Samba is disabled."
         );
```

I don't think this change was erroneous, after all if we remove the new duplicate `cfg.enable` we end up with an incorrect comment
```suggestion
        environment.etc."samba/smb.conf".source = mkOptionDefault configFile;
```

This suggest to me that 7435db4f898233f9615b7818c07bbbcf30d44d63 was a fix for dead-code due to incorrect nesting.